src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.Editor.cs

```diff
@@ -370,8 +384,9 @@ private async Task<(Document, bool addedFields)> GenerateFieldDelegatingConstruc
                                 }
                                 else
                                 {
-                                    // Can change the parameter name, so do so.
-                                    var newParameterName = new ParameterName(newFieldName, isFixed: false);
+                                    // Can change the parameter name, so do so.  
+                                    // But first remove any prefix added due to field naming styles
+                                    var newParameterName = new ParameterName(newFieldName.Substring(fieldNamingRule.NamingStyle.Prefix.Length), isFixed: false);
```

Comment:
❓ Why not use the effective parameter naming style here?

Reply:
I implemented that.